src/components/accessories/patientOPD/patientOPD.tsx

```diff
@@ -89,17 +90,34 @@ const PatientOPD: FunctionComponent = () => {
     }
   }, [dispatch, activityTransitionState]);
 
-  const onSubmit = (opdValues: OpdWithOperationRows) => {
+  const onSubmit = (opdValues: OpdWithOperatioRowDTO) => {
     setShouldResetForm(false);
-    opdValues.opd.patientCode = patient?.code;
-    opdValues.opd.age = patient?.age;
-    opdValues.opd.sex = patient?.sex;
-    opdValues.opd.userID = userId;
-    const opdToSave = { ...opdToEdit, ...opdValues.opd };
-    if (!creationMode && opdToEdit.code) {
-      dispatch(updateOpd(opdToEdit.code, opdToSave, opdValues.operationRows));
+    opdValues.opdDTO.patientCode = patient?.code;
+    opdValues.opdDTO.age = patient?.age;
+    opdValues.opdDTO.sex = patient?.sex;
+    opdValues.opdDTO.userID = userId;
+    opdValues.opdDTO.patientName =
+      patient?.firstName + " " + patient?.secondName;
+    const opdToSave = { ...opdToEdit.opdDTO, ...opdValues.opdDTO };
+    if (!creationMode && opdToEdit.opdDTO.code) {
```

Comment:
I can't understand the purposes of this condition.

Reply:
We will have syntax error in case we dont put the not null condition on code.